Api/Services/EventService.cs

```diff
@@ -243,6 +243,134 @@ public async Task<Result<Pagination<EventSummaryVM>>> GetEventsInterestedInAsync
 
             return await query.PaginateAsync(page, perPage);
         }
-    }
 
+
+        /// <summary>
+        /// Updates the details of an existing event.
+        /// </summary>
+        /// <param name="eventId">The id of the event to update.</param>
+        /// <param name="request">The new details for the event.</param>
+        /// <param name="user">The user updating the event.</param>
+        /// <returns>A Result object containing the updated event or validation failures.</returns>
+        public async Task<Result<EventVM>> UpdateEventAsync(int eventId, UpdateEventRequest request, User user)
+        {
+           var eventToUpdate = await context.Events
+            .Include(e => e.Creator)
+            .Include(e => e.Restaurant)
+            .Include(e => e.Interested)
+            .FirstOrDefaultAsync(e => e.Id == eventId);
+
+            if (eventToUpdate is null)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = null,
+                    ErrorCode = ErrorCodes.NotFound,
+                    ErrorMessage = ErrorCodes.NotFound
+                };
+            }
+
+            if(eventToUpdate.Creator!=user)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = null,
+                    ErrorCode = ErrorCodes.AccessDenied,
+                    ErrorMessage = ErrorCodes.AccessDenied
+                };
+            }
+
+            if (eventToUpdate.Time < DateTime.UtcNow)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = null,
+                    ErrorCode = ErrorCodes.DateMustBeInFuture,
+                    ErrorMessage = ErrorCodes.DateMustBeInFuture
+                };
+            }
+
+            eventToUpdate.Description = request.Description;
+            eventToUpdate.Time = request.Time;
+            eventToUpdate.MustJoinUntil = request.MustJoinUntil;
+            eventToUpdate.RestaurantId = request.RestaurantId;
+
+            var restaurant = await context.Restaurants.FindAsync(request.RestaurantId);
+            if (restaurant is null)
+            {
+                return new ValidationFailure
+                {
+                    PropertyName = nameof(request.RestaurantId),
+                    ErrorCode = ErrorCodes.RestaurantDoesNotExist,
+                    ErrorMessage = ErrorCodes.RestaurantDoesNotExist
+                };
+            }
```

Comment:
to trzeba sprawdzać zanim zmienimy `eventToUpdate`, bo inaczej te zmiany mogą być zapisane

Reply:
Moje przeoczenie, ale chyba i tak by się nie zapisało bez tego:             await context.SaveChangesAsync();
tak czy inaczej naprawiłem